test/server/connection_handler_test.cc

```diff
@@ -787,7 +803,7 @@ TEST_F(ConnectionHandlerTest, FallbackToWildcardListener) {
   EXPECT_CALL(*access_log_, log(_, _, _, _));
 }
 
-TEST_F(ConnectionHandlerTest, OldBehaviorMatchFirstWildcardListener) {
```

Comment:
A note for the reviewer, the original test ensure the first listener matched since the original code using the `list`. After using the unorder map, so this behavior changed. I assume the `MatchFirst` isn't part of API contract from the doc.